persistent-postgresql/test/PgIntervalTest.hs

```diff
@@ -0,0 +1,53 @@
+{-# LANGUAGE EmptyDataDecls             #-}
+{-# LANGUAGE FlexibleContexts           #-}
+{-# LANGUAGE GADTs                      #-}
+{-# LANGUAGE GeneralizedNewtypeDeriving #-}
+{-# LANGUAGE MultiParamTypeClasses      #-}
+{-# LANGUAGE OverloadedStrings          #-}
+{-# LANGUAGE QuasiQuotes                #-}
+{-# LANGUAGE TemplateHaskell            #-}
+{-# LANGUAGE TypeFamilies               #-}
+{-# LANGUAGE UndecidableInstances       #-}
+{-# LANGUAGE DerivingStrategies         #-}
+{-# LANGUAGE StandaloneDeriving         #-}
+{-# LANGUAGE DeriveAnyClass             #-}
+
+module PgIntervalTest where
+
+import PgInit
+import Control.Monad.IO.Unlift
+import Database.Persist.Postgresql (PgInterval(..))
+
+share [mkPersist sqlSettings, mkMigrate "pgIntervalMigrate"] [persistLowerCase|
+PgIntervalDb
+    interval_field PgInterval
+    deriving Eq
+    deriving Show
+|]
+
+specs :: (MonadUnliftIO m, MonadFail m) => RunDb SqlBackend m -> Spec
+specs runDb = do
+  describe "Postgres Interval tests" $ do
```

Comment:
These round-trip tests are a great candidate for `QuickCheck`. We can write:

```haskell
describe "pg interval" $ do
  prop "round trips" $ \time -> runDb $ do
    let example = PgIntervalDb $ PgInterval time
    rid <- insert example
    r <- getJust rid
    liftIO $ r `shouldBe` example
```

and it should catch edge casees.